plugins/modules/nmcli.py

```diff
@@ -1584,10 +1627,11 @@ def connection_options(self, detect_change=False):
         if self.mtu_conn_type:
             options.update({self.mtu_setting: self.mtu})
 
-        # Connections that can have a master.
-        if self.slave_conn_type:
+        # NetworkManager allows any interface type to be slave
+        if self.master:
             options.update({
                 'connection.master': self.master,
+                'connection.slave-type': self.slave_type,
```

Comment:
```connection.slave-type``` will be rewritten if you use ```bond-slave```, ```bridge-slave```, ```team-slave``` as a type. I guess it's better to make them mutually exclusive or show some kind of warning

Reply:
Agree, will update soon

Reply:
Added several module fails and warning messages about it.